arboretum/permissions/checks/test_org_permissions.py

```diff
@@ -0,0 +1,119 @@
+# -*- mode:python; coding:utf-8 -*-
+# Copyright (c) 2021 IBM Corp. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repository organization permissions check."""
+
+from compliance.check import ComplianceCheck
+from compliance.evidence import DAY, ReportEvidence, evidences
+from compliance.utils.data_parse import get_sha256_hash
+
+
+class OrgPermissionsCheck(ComplianceCheck):
+    """Checks for repository organization/owner permissions."""
+
+    @property
+    def title(self):
+        """
+        Return the title of the checks.
+
+        :returns: the title of the checks
+        """
+        return 'Repository Organization/Owner Permissions'
+
+    @classmethod
+    def setUpClass(cls):
+        """Initialize the check object with configuration settings."""
+        cls.config.add_evidences(
+            [
+                ReportEvidence(
+                    'org_permissions.md',
+                    'permissions',
+                    DAY,
+                    'Repository organization permission report.'
+                )
+            ]
+        )
+        return cls
+
+    def test_org_permissions(self):
+        """Check the access to organization repositories."""
+        orgs = self.config.get('org.permissions.org_integrity.orgs')
+        for org in orgs:
+            host, org_name = org['url'].rsplit('/', 1)
+            service = 'gh'
+            if 'gitlab' in host:
+                service = 'gl'
+            elif 'bitbucket' in host:
+                service = 'bb'
+            url_hash = get_sha256_hash([org['url']], 10)
+            evidence_paths = {
+                'direct': f'raw/permissions/{service}_direct_collaborators_'
+                + f'{url_hash}.json',
+                'outside': f'raw/permissions/{service}_outside_collaborators_'
+                + f'{url_hash}.json',
+                'forks': f'raw/permissions/{service}_forks_{url_hash}.json'
+            }
```

Comment:
This is pretty ugly and hard to read.  If you renamed some things to get this to fit 80 char then you could rewrite this as:

```suggestion
            path = 'raw/permissions/{}_{}_{}.json'
            evs = ['direct_collaborators', 'outside_collaborators', 'forks']
            ev_paths = {ev: path.format(service, ev, url_hash) for ev in evs}
```

`evidence_paths` would now be `ev_paths` and its keys would be 'direct_collaborators', 'outside_collaborators', 'forks'  but I think this is better.

Reply:
Done